datafusion/physical-plan/src/joins/hash_join.rs

```diff
@@ -566,8 +566,87 @@ impl HashJoinExec {
             boundedness_from_children([left, right]),
         ))
     }
+
+    /// Returns a new `ExecutionPlan` that computes the same join as this one,
+    /// with the left and right inputs swapped using the  specified
+    /// `partition_mode`.
+    ///
+    /// # Notes:
+    ///
+    /// This function is public so other downstream projects can use it to
+    /// construct `HashJoinExec` with right side as the build side.
+    pub fn swap_inputs(
+        &self,
+        partition_mode: PartitionMode,
```

Comment:
Isn't this a parameter which is present in the HashJoinExec itself? Why are we separately sending it 🤔

Reply:
I am not 100% sure

Some uses clearly are unnecessary where the type of the join is simply passed in: https://github.com/apache/datafusion/blob/fce3fb3033ae50e8590e075ffb11422112128233/datafusion/core/src/physical_optimizer/join_selection.rs#L569

Other places like this are less clear that the output type is the same as the input type:
https://github.com/apache/datafusion/blob/fce3fb3033ae50e8590e075ffb11422112128233/datafusion/core/src/physical_optimizer/join_selection.rs#L285

I'll try and make a PR and see what happens if we don't send it

Reply:
- I made https://github.com/apache/datafusion/pull/13940

the sqllogictests seem to have passed. I'll check the rest of CI in a bit